staging/src/k8s.io/apiserver/pkg/registry/generic/registry/storage_factory.go

```diff
@@ -83,15 +84,21 @@ func StorageWithCacher() generic.StorageDecorator {
 	}
 }
 
-func objectTypeToString(obj runtime.Object) string {
+func objectTypeToString(obj runtime.Object) []interface{} {
+	types := []string{"type", fmt.Sprintf("%T", obj)}
 	// special-case unstructured objects that tell us their apiVersion/kind
 	if u, isUnstructured := obj.(*unstructured.Unstructured); isUnstructured {
 		if apiVersion, kind := u.GetAPIVersion(), u.GetKind(); len(apiVersion) > 0 && len(kind) > 0 {
-			return fmt.Sprintf("apiVersion=%s, kind=%s", apiVersion, kind)
+			types = []string{"apiVersion", apiVersion, "kind", kind}
 		}
 	}
-	// otherwise just return the type
-	return fmt.Sprintf("%T", obj)
+
+	var ret []interface{}
```

Comment:
Could we skip this loop by changing `types` slice type?

Reply:
modified.